<!DOCTYPE html><html><head><title>R: ARTP test for multiple independent populations</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table style="width: 100%;"><tr><td>sARTP.multiPop.SNP.union {ARTP2}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id='sARTP.multiPop.SNP.union'>ARTP test for multiple independent populations</h2>

<h3>Description</h3>

<p>Calculate gene and pathway p-values using the ARTP test and summary data from multiple independent populations. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sARTP.multiPop.SNP.union(summary.files.list, pathway, family, reference.list, lambda.list, 
      ncases.list, ncontrols.list, nsamples.list, options.list=NULL, options.merged=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sARTP.multiPop.SNP.union_:_summary.files.list">summary.files.list</code></td>
<td>

<p>A list of summary files for all populations. Each element of the list should correspond to the 
summary files for one population. 
See the documentation for <code>summary.files</code> in <code><a href="../../ARTP2/help/sARTP.html">sARTP</a></code>.
</p>
</td></tr>
<tr><td><code id="sARTP.multiPop.SNP.union_:_pathway">pathway</code></td>
<td>

<p>See the documentation for <code>pathway</code> in <code><a href="../../ARTP2/help/sARTP.html">sARTP</a></code>.
</p>
</td></tr>
<tr><td><code id="sARTP.multiPop.SNP.union_:_family">family</code></td>
<td>

<p>A string taking values of <code>'gaussian'</code> or <code>'binomial'</code>. 
</p>
</td></tr>
<tr><td><code id="sARTP.multiPop.SNP.union_:_reference.list">reference.list</code></td>
<td>

<p>A list of reference sets for all populations. Each element of the list should correspond to the 
reference set for one population. 
See the documentation for <code>reference</code> in <code><a href="../../ARTP2/help/sARTP.html">sARTP</a></code>.
</p>
</td></tr>
<tr><td><code id="sARTP.multiPop.SNP.union_:_lambda.list">lambda.list</code></td>
<td>

<p>A list of lambda values for all populations. Each element of the list should correspond to the 
lambda values for one population. 
See the documentation for <code>lambda</code> in <code><a href="../../ARTP2/help/sARTP.html">sARTP</a></code>.
</p>
</td></tr>
<tr><td><code id="sARTP.multiPop.SNP.union_:_ncases.list">ncases.list</code></td>
<td>

<p>NULL or a list of the number of cases for all populations. Each element of the list should correspond to the 
number of cases for one population. 
See the documentation for <code>ncases</code> in <code><a href="../../ARTP2/help/sARTP.html">sARTP</a></code>.
</p>
</td></tr>
<tr><td><code id="sARTP.multiPop.SNP.union_:_ncontrols.list">ncontrols.list</code></td>
<td>

<p>NULL or a list of the number of controls for all populations. Each element of the list should correspond to the 
number of controls for one population. 
See the documentation for <code>ncontrols</code> in <code><a href="../../ARTP2/help/sARTP.html">sARTP</a></code>.
</p>
</td></tr>
<tr><td><code id="sARTP.multiPop.SNP.union_:_nsamples.list">nsamples.list</code></td>
<td>

<p>NULL or a list of the number of samples for all populations. Each element of the list should correspond to the 
number of samples for one population. 
See the documentation for <code>nsamples</code> in <code><a href="../../ARTP2/help/sARTP.html">sARTP</a></code>.
</p>
</td></tr>
<tr><td><code id="sARTP.multiPop.SNP.union_:_options.list">options.list</code></td>
<td>

<p>NULL or a list of <code><a href="../../ARTP2/help/options.html">options</a></code> for all populations. Each element of the list should correspond to the 
<code><a href="../../ARTP2/help/options.html">options</a></code> for one population. 
See <code><a href="../../ARTP2/help/options.html">options</a></code>.
</p>
</td></tr>
<tr><td><code id="sARTP.multiPop.SNP.union_:_options.merged">options.merged</code></td>
<td>
<p> NULL or a list of options for the merged population.
See <code><a href="../../ARTP2/help/options.html">options</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is similar to <code><a href="../../ARTP2/help/sARTP.html">sARTP</a></code> except that it deals with multiple populations
instead of just a single population. All the populations should be independent.
The algorithm for this function is as follows: <br />
1. Loop over each population and perform the SNP/gene filtering based on 
each population's reference set and options. <br />
2. Define the pathway to be the union of all SNPs remaining after the filtering in step 1. <br />
3. Define the groups from the pathway defined in step 2. <br />
4. For each population, recover the summary statistics (score and covariance matrix) within each group. <br />
5. Merge the summary statistics together across all populations for each group. <br />
6. Generate permuted test statistics for the merged population. <br />
7. Compute the gene and pathway p-values for the merged population.
</p>
<p><b>Handwritten notes on the algorithm are at .../inst/doc/SNP_based_pathway_union.pdf</b>
</p>


<h3>Value</h3>

<p>A list containing the following components: 
</p>
<table>
<tr><td><code>pathway.pvalue</code></td>
<td>
<p>final pathway p-value accounting for multiple comparisons.</p>
</td></tr>
<tr><td><code>gene.pvalue</code></td>
<td>
<p>a data frame containing gene name, number of SNPs in the gene that were included in the analysis, chromosome name, and the p-value for the gene accounting for multiple comparisons.</p>
</td></tr>
<tr><td><code>pathway</code></td>
<td>
<p>a data frame defining the pathway that was actually tested after various filters applied. </p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>a list containing detailed information of selected SNPs in each gene. </p>
</td></tr>
<tr><td><code>most.sig.genes</code></td>
<td>
<p>a character vector of genes selected by <code>ARTP2</code>. They are the most promising candidates, although their statistical significance is not guaranteed. </p>
</td></tr>
<tr><td><code>options</code></td>
<td>
<p>a list of options used in the analysis. See <code><a href="../../ARTP2/help/options.html">options</a></code></p>
</td></tr>
<tr><td><code>pathway.nperm</code></td>
<td>
<p>The number of permutations for computing the pathway p-value. This takes
into account the option <code>expand.fold</code>.</p>
</td></tr>
<tr><td><code>accurate</code></td>
<td>
<p><code>TRUE</code> if <code>options$nperm</code> is large enougth to accurately estimate p-values,
i.e., if the criteria <br /> 
<code>(sqrt(pvalue*(1-pvalue)/pathway.nperm)/pvalue &lt; 0.1) &amp;&amp; </code> <br />
<code>(ceiling(-log10(minimum gene pvalue)) &lt; log10(nperm))</code> <br />
is satisfied.
</p>
</td></tr>
<tr><td><code>setup</code></td>
<td>
<p>a list containing necessary input for <code><a href="../../ARTP2/help/warm.start.multiPop.SNP.union.html">warm.start.multiPop.SNP.union</a></code>. 
It can be written to a file by using the function <code><a href="../../base/help/save.html">save</a></code>, 
then its path can be the input of <code><a href="../../ARTP2/help/warm.start.multiPop.SNP.union.html">warm.start.multiPop.SNP.union</a></code>. 
</p>
</td></tr>
</table>


<h3>References</h3>

<p>Zhang H, Wheeler W, Hyland LP, Yang Y, Shi J, Chatterjee N, Yu K. (2016) A powerful procedure for pathway-based meta-analysis using summary statistics identifies 43 pathways associated with type II diabetes in European populations. PLoS Genetics 12(6): e1006122
</p>
<p>Yu K, Li Q, Bergen AW, Pfeiffer RM, Rosenberg PS, Caporaso N, Kraft P, Chatterjee N. (2009) Pathway analysis by adaptive combination of P-values. Genet Epidemiol 33(8): 700 - 709
</p>
<p>Zhang H, Shi J, Liang F, Wheeler W, Stolzenberg-Solomon R, Yu K.  (2014) A fast multilocus test with adaptive SNP selection for large-scale genetic association studies. European Journal of Human Genetics 22: 696 - 702
</p>


<h3>See Also</h3>

<p><code><a href="../../ARTP2/help/options.html">options</a></code>, <code><a href="../../ARTP2/help/sARTP.html">sARTP</a></code>, <code><a href="../../ARTP2/help/warm.start.multiPop.SNP.union.html">warm.start.multiPop.SNP.union</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  library(ARTP2)

  # Pathway definition
  pathway &lt;- system.file("extdata", package = "ARTP2", "pathway.txt.gz")

  # logistic regression is used in base model
  family &lt;- 'binomial'

  ## Create data frame containing paths of PLINK files that are going to used as the reference
  ## As a small example, use chromosomes 20 and 22
  chr  &lt;- c(20, 22)
  nchr &lt;- length(chr)

  # Define reference data for each population
  fam &lt;- vector("character", nchr)
  bim &lt;- vector("character", nchr)
  bed &lt;- vector("character", nchr)
  for(i in 1:nchr){
    fam[i] &lt;- system.file("extdata", package = "ARTP2", paste("chr", chr[i], ".fam", sep = ""))
    bim[i] &lt;- system.file("extdata", package = "ARTP2", paste("chr", chr[i], ".bim", sep = ""))
    bed[i] &lt;- system.file("extdata", package = "ARTP2", paste("chr", chr[i], ".bed", sep = ""))
  }
  reference1 &lt;- data.frame(fam, bim, bed, stringsAsFactors = FALSE)
  for(i in 1:nchr){
    fam[i] &lt;- system.file("extdata", package = "ARTP2", paste("chr", chr[i], "_EAS.fam", sep = ""))
    bim[i] &lt;- system.file("extdata", package = "ARTP2", paste("chr", chr[i], "_EAS.bim", sep = ""))
    bed[i] &lt;- system.file("extdata", package = "ARTP2", paste("chr", chr[i], "_EAS.bed", sep = ""))
  }
  reference2 &lt;- data.frame(fam, bim, bed, stringsAsFactors = FALSE)

  ## Other objects for population 1
  study1     &lt;- system.file("extdata", package = "ARTP2", "study1.txt.gz")
  ncases1    &lt;- list(c(9580, 2591))
  ncontrols1 &lt;- list(c(53810, 3052))
  lambda1    &lt;- 1.10
  options1   &lt;- list(maf = .01, HWE.p = 1e-6, 
                     id.str = "Pop1", out.dir=getwd(), 
                     save.setup = FALSE)

  # For population 2
  study2     &lt;- system.file("extdata", package = "ARTP2", "study_EAS.txt.gz")
  ncases2    &lt;- list(1000)
  ncontrols2 &lt;- list(1000)
  lambda2    &lt;- 1.13
  options2   &lt;- list(maf = .01, HWE.p = 1e-6,  
                     id.str = "Pop2", out.dir=getwd(), 
                     save.setup = FALSE)

  #ret &lt;- sARTP.multiPop.SNP.union(list(study1, study2), pathway, family, list(reference1, reference2), 
  #                      list(lambda1, lambda2), ncases.list=list(ncases1, ncases2), 
  #                      ncontrols.list=list(ncontrols1, ncontrols2), 
  #                      options.list=list(options1, options2), options.merged=NULL)

</code></pre>

<hr /><div style="text-align: center;">[Package <em>ARTP2</em> version 1.2.32 <a href="00Index.html">Index</a>]</div>
</div>
</body></html>
