<!DOCTYPE html><html><head><title>R: Performing ARTP2 test with warm start</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table style="width: 100%;"><tr><td>warm.start {ARTP2}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id='warm.start'>
Performing ARTP2 test with warm start
</h2>

<h3>Description</h3>

<p>This function is designed to accelerate the <code>ARTP2</code> test in practice. It uses pre-calculated and reusable statistics as input and allow the users to try different testing configuration more efficiently. See <code>Details</code> for more information. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>warm.start(setup, nperm=NULL, lambda=1.0, nthread=NULL, expand.fold=0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="warm.start_:_setup">setup</code></td>
<td>
<p>an R object created by <code><a href="../../ARTP2/help/sARTP.html">sARTP</a></code> or <code><a href="../../ARTP2/help/rARTP.html">rARTP</a></code>. It is a list containing necessary statistics for computing p-values. </p>
</td></tr>
<tr><td><code id="warm.start_:_nperm">nperm</code></td>
<td>
<p>the number of permutations. If it is <code>NULL</code> as default, then the value in the saved setup will be used. See <code>Details</code>. </p>
</td></tr>
<tr><td><code id="warm.start_:_lambda">lambda</code></td>
<td>
<p>inflation factor to be adjusted in pathway analysis. <code>lambda</code> in this function can only be a single numeric number, which is different from the one in <code>sARTP</code>. The default is 1.0. </p>
</td></tr>
<tr><td><code id="warm.start_:_nthread">nthread</code></td>
<td>
<p>number of threads to be used in permutation. <code>NULL</code> if <code>setup$options$nthread</code> is used. </p>
</td></tr>
<tr><td><code id="warm.start_:_expand.fold">expand.fold</code></td>
<td>
<p>The number of copies of the <code>nperm</code> permutations to expand.
Setting this option to a positive integer will expand the number of
permutations by randomly permuting the <code>nperm</code> permutations
within each group <code>expand.fold</code> times when computing the 
pathway p-value. The pathway p-value is then essentially computed
using <code>(1 + expand.fold)*nperm</code> permutations. 
The default is 0.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>An <code>ARTP2</code> test has two major steps in testing an association. The first step applys data clean criteria and creates necessary and reusable statistics, which can be time-consuming for large pathways. The second step performs the testing procedure to estimate the pathway or gene-level p-value. <code>warm.start</code> focuses on the second step. 
</p>
<p>The first step can be done by using <code><a href="../../ARTP2/help/sARTP.html">sARTP</a></code> or <code><a href="../../ARTP2/help/rARTP.html">rARTP</a></code> if their <code>options$only.setup</code> is set as <code>TRUE</code>. Their output object, <code>setup</code>, can be used as the first argument of <code>warm.start</code>. With <code>warm.start</code>, users can try different configurations to perform various tests allowed by the <code>ARTP2</code> framework, but avoid long waiting time for data cleaning. Commonly used options in <code>setup$options</code> include <code>method</code>, <code>inspect.snp.n</code>, <code>inspect.gene.n</code>, <code>nperm</code>, etc. 
</p>
<p>Note that both <code><a href="../../ARTP2/help/sARTP.html">sARTP</a></code> and <code><a href="../../ARTP2/help/rARTP.html">rARTP</a></code> can produce the final p-value directly if <code>options$only.setup</code> is <code>FALSE</code>. 
</p>
<p>The <code>setup</code> is supposed to have all components defined in <code><a href="../../ARTP2/help/sARTP.html">sARTP</a></code> and <code><a href="../../ARTP2/help/rARTP.html">rARTP</a></code>. If <code>nperm</code> is <code>NULL</code>, then it will be set as <code>setup$options$nperm</code>. The users can also pass <code>lambda</code> if a second round genomic control is needed. However, unlike in <code><a href="../../ARTP2/help/sARTP.html">sARTP</a></code>, <code>lambda</code> here can only be a single numberic number rather than a vector. Options <code>nperm</code> and <code>lambda</code> are the most useful ones in using <code>warm.start</code> so we highlight them in the interface. Users can modify any option values in <code>setup$options</code> directly to get more controls of the testing procedure. See <code><a href="../../ARTP2/help/options.html">options</a></code> for more details about how to set <code>setup$options</code>. 
</p>
<p>Except for <code>setup$options</code>, all other components in <code>setup</code> should not be modified by users. 
</p>


<h3>Value</h3>

<p><code>warm.start</code> returns an object of class <code>ARTP2</code>. It is a list containing the following components: 
</p>
<table>
<tr><td><code>pathway.pvalue</code></td>
<td>
<p>final pathway p-value accounting for multiple comparisons.</p>
</td></tr>
<tr><td><code>gene.pvalue</code></td>
<td>
<p>a data frame containing gene name, number of SNPs in the gene that were included in the analysis, chromosome name, and the p-value for the gene accounting for multiple comparisons.</p>
</td></tr>
<tr><td><code>pathway</code></td>
<td>
<p>a data frame defining the pathway that was actually tested after various filters applied. </p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>a list containing detailed information of selected SNPs in each gene. </p>
</td></tr>
<tr><td><code>most.sig.genes</code></td>
<td>
<p>a character vector of genes selected by <code>ARTP2</code>. They are the most promising candidates, although their statistical significance is not guaranteed. </p>
</td></tr>
<tr><td><code>deleted.snps</code></td>
<td>
<p>a data frame containing SNPs excluded from the analysis and their reasons.</p>
</td></tr>
<tr><td><code>deleted.genes</code></td>
<td>
<p>a data frame containing genes excluded from the analysis because they are subsets of other remaining genes. </p>
</td></tr>
<tr><td><code>options</code></td>
<td>
<p>a list of options used in the analysis. See <code><a href="../../ARTP2/help/options.html">options</a></code></p>
</td></tr>
<tr><td><code>test.timing</code></td>
<td>
<p>timing information (in sec)</p>
</td></tr>
<tr><td><code>pathway.nperm</code></td>
<td>
<p>The number of permutations for computing the pathway p-value. This takes
into account the option <code>expand.fold</code>.</p>
</td></tr>
<tr><td><code>accurate</code></td>
<td>
<p><code>TRUE</code> if <code>options$nperm</code> is large enougth to accurately estimate p-values,
i.e., if the criteria <br /> 
<code>(sqrt(pvalue*(1-pvalue)/pathway.nperm)/pvalue &lt; 0.1) &amp;&amp; </code> <br />
<code>(ceiling(-log10(minimum gene pvalue)) &lt; log10(nperm))</code> <br />
is satisfied.
</p>
</td></tr>
</table>


<h3>References</h3>

<p>Zhang H, Wheeler W, Hyland LP, Yang Y, Shi J, Chatterjee N, Yu K. (2016) A powerful procedure for pathway-based meta-analysis using summary statistics identifies 43 pathways associated with type II diabetes in European populations. PLoS Genetics 12(6): e1006122
</p>
<p>Yu K, Li Q, Bergen AW, Pfeiffer RM, Rosenberg PS, Caporaso N, Kraft P, Chatterjee N. (2009) Pathway analysis by adaptive combination of P-values. Genet Epidemiol 33(8): 700 - 709
</p>
<p>Zhang H, Shi J, Liang F, Wheeler W, Stolzenberg-Solomon R, Yu K.  (2014) A fast multilocus test with adaptive SNP selection for large-scale genetic association studies. European Journal of Human Genetics: 22, 696 - 702
</p>


<h3>See Also</h3>

<p><code><a href="../../ARTP2/help/sARTP.html">sARTP</a></code>, <code><a href="../../ARTP2/help/rARTP.html">rARTP</a></code>, <code><a href="../../ARTP2/help/options.html">options</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## firstly, run the example in sARTP
## users can adjust the second round inflation in warm.start
## the first round inflation can be study-specific and adjusted in rARTP 
## or sARTP

library(ARTP2)
study1 &lt;- system.file("extdata", package = "ARTP2", "study1.txt.gz")
study2 &lt;- system.file("extdata", package = "ARTP2", "study2.txt.gz")
pathway &lt;- system.file("extdata", package = "ARTP2", "pathway.txt.gz")
chr &lt;- 1:22
nchr &lt;- length(chr)
fam &lt;- vector("character", nchr)
bim &lt;- vector("character", nchr)
bed &lt;- vector("character", nchr)
for(i in 1:nchr){
  fam[i] &lt;- system.file("extdata", package = "ARTP2", paste("chr", chr[i], ".fam", sep = ""))
  bim[i] &lt;- system.file("extdata", package = "ARTP2", paste("chr", chr[i], ".bim", sep = ""))
  bed[i] &lt;- system.file("extdata", package = "ARTP2", paste("chr", chr[i], ".bed", sep = ""))
}
reference &lt;- data.frame(fam, bim, bed)
options &lt;- list(inspect.snp.n = 2, nperm = 1e4, 
                maf = .01, HWE.p = 1e-6, 
                gene.R2 = .9, 
                id.str = "unique-pathway-id", 
                out.dir = getwd(), save.setup = FALSE)
                
## different inflation factors are adjusted in two studies
## first round adjustment
lambda &lt;- c(1.10, 1.08)
ncases &lt;- list()
ncontrols &lt;- list()
ncases[[1]] &lt;- c(9580, 2591)
ncontrols[[1]] &lt;- c(53810, 3052)
ncases[[2]] &lt;- 7638
ncontrols[[2]] &lt;- 54319

family &lt;- 'binomial'

## do not run permutation
options$only.setup &lt;- TRUE
## the first round study-specific inflation is adjusted as lambda = c(1.10, 1.08)
# setup &lt;- sARTP(summary.files = c(study1, study2), pathway, family, reference, lambda, 
#                ncases, ncontrols, options = options)

## the two rounds of inflation is adjusted as lambda2 = c(1.17370, 1.15236)
lambda2 &lt;- lambda * 1.067
## run permutation to calculate p-value
options$only.setup &lt;- FALSE
# ret1 &lt;- sARTP(summary.files = c(study1, study2), pathway, family, reference, lambda2, 
#               ncases, ncontrols, options = options)

## or adjust the second round of inflation in warm.start
# ret2 &lt;- warm.start(setup, lambda = 1.067)

# two ways of inflation adjustment should give same results
# ret1$pathway.pvalue == ret2$pathway.pvalue

###############################################################
###############################################################
## modify or specify the method
# setup$options$method &lt;- 2
# setup$options$inspect.snp.n &lt;- 3

## nthread = 2 for Linux only
## nthread will be reset to 1 under Windows and Mac OS
# ret3 &lt;- warm.start(setup, nperm = 1e5, nthread = 2)

</code></pre>

<hr /><div style="text-align: center;">[Package <em>ARTP2</em> version 1.2.32 <a href="00Index.html">Index</a>]</div>
</div>
</body></html>
