<!DOCTYPE html><html><head><title>R: ARTP2 for multiple independent populations</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table style="width: 100%;"><tr><td>warm.start.multiPop.gene {ARTP2}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2 id='warm.start.multiPop.gene'>
ARTP2 for multiple independent populations
</h2>

<h3>Description</h3>

<p>To perform the ARTP2 test by merging gene results from multiple independent populations. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>warm.start.multiPop.gene(sARTP2.list, method="FISHER", nthread=1, delete.gene.files=TRUE,
                         expand.fold=0, use.ranks.pathway=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="warm.start.multiPop.gene_:_sARTP2.list">sARTP2.list</code></td>
<td>
<p>A list of return objects from <code><a href="../../ARTP2/help/sARTP.html">sARTP</a></code> or .rda files 
containing the returned objects from <code><a href="../../ARTP2/help/sARTP.html">sARTP</a></code> 
for each population.</p>
</td></tr>
<tr><td><code id="warm.start.multiPop.gene_:_method">method</code></td>
<td>
<p>One of &quot;MAX&quot;, &quot;META&quot;, &quot;FISHER&quot;, &quot;ACAT&quot; or &quot;WGTFISHER&quot; for the method
to combine the permuted test statistics from each population.
The default is &quot;FISHER&quot;.</p>
</td></tr>
<tr><td><code id="warm.start.multiPop.gene_:_nthread">nthread</code></td>
<td>
<p>Number of threads to be used. The default is 1.</p>
</td></tr>
<tr><td><code id="warm.start.multiPop.gene_:_delete.gene.files">delete.gene.files</code></td>
<td>
<p>TRUE or FALSE to delete the temporary gene files from each population. 
The default is TRUE. </p>
</td></tr>
<tr><td><code id="warm.start.multiPop.gene_:_expand.fold">expand.fold</code></td>
<td>
<p>The number of copies of the <code>nperm</code> permutations to expand.
Setting this option to a positive integer will expand the number of
permutations by randomly permuting the <code>nperm</code> permutations
within each gene <code>expand.fold</code> times when computing the 
pathway p-value. The pathway p-value is then essentially computed
using <code>(1 + expand.fold)*nperm</code> permutations. 
The default is 0.</p>
</td></tr>
<tr><td><code id="warm.start.multiPop.gene_:_use.ranks.pathway">use.ranks.pathway</code></td>
<td>
<p>TRUE or FALSE to transform the merged gene p-values into
permutation p-values based on their ranks. This option only affects
the pathway p-value and is only applied 
when <code>expand.fold = 0</code>.
The default is TRUE. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function combines the gene p-values from the populations using Fisher's method for all permutations,
where the number of permutations is the minimum from the populations. Note that if more
permutations are needed, then each population must be first run with a greater number
of permutations. 
If a particular gene is only present in a subset of the populations, then the combined gene p-value is
from the subset of populations.
</p>
<p><b>When <code><a href="../../ARTP2/help/sARTP.html">sARTP</a></code> is called for each population, 
the option <code>options$delete.gene.files</code> must be set to FALSE, and
a different seed should be passed in using <code>options$seed</code>.</b>
</p>


<h3>Value</h3>

<p>A list containing the following components: 
</p>
<table>
<tr><td><code>pathway.pvalue</code></td>
<td>
<p>final pathway p-value accounting for multiple comparisons.</p>
</td></tr>
<tr><td><code>gene.pvalue</code></td>
<td>
<p>a data frame containing gene name, number of SNPs in the gene that were included in the analysis, chromosome name, and the p-value for the gene accounting for multiple comparisons.</p>
</td></tr>
<tr><td><code>pathway</code></td>
<td>
<p>a data frame defining the pathway that was actually tested after various filters applied. </p>
</td></tr>
<tr><td><code>most.sig.genes</code></td>
<td>
<p>a character vector of genes selected by <code>ARTP2</code>. They are the most promising candidates, although their statistical significance is not guaranteed. </p>
</td></tr>
<tr><td><code>test.timing</code></td>
<td>
<p>timing information (in sec)</p>
</td></tr>
<tr><td><code>pathway.nperm</code></td>
<td>
<p>The number of permutations for computing the pathway p-value. This takes
into account the option <code>expand.fold</code>.</p>
</td></tr>
<tr><td><code>accurate</code></td>
<td>
<p><code>TRUE</code> if <code>options$nperm</code> is large enougth to accurately estimate p-values,
i.e., if the criteria <br /> 
<code>(sqrt(pvalue*(1-pvalue)/pathway.nperm)/pvalue &lt; 0.1) &amp;&amp; </code> <br />
<code>(ceiling(-log10(minimum gene pvalue)) &lt; log10(nperm))</code> <br />
is satisfied.
</p>
</td></tr>
</table>


<h3>References</h3>

<p>Zhang H, Wheeler W, Hyland LP, Yang Y, Shi J, Chatterjee N, Yu K. (2016) A powerful procedure for pathway-based meta-analysis using summary statistics identifies 43 pathways associated with type II diabetes in European populations. PLoS Genetics 12(6): e1006122
</p>
<p>Yu K, Li Q, Bergen AW, Pfeiffer RM, Rosenberg PS, Caporaso N, Kraft P, Chatterjee N. (2009) Pathway analysis by adaptive combination of P-values. Genet Epidemiol 33(8): 700 - 709
</p>
<p>Zhang H, Shi J, Liang F, Wheeler W, Stolzenberg-Solomon R, Yu K.  (2014) A fast multilocus test with adaptive SNP selection for large-scale genetic association studies. European Journal of Human Genetics: 22, 696 - 702
</p>


<h3>See Also</h3>

<p><code><a href="../../ARTP2/help/sARTP.html">sARTP</a></code>, <code><a href="../../ARTP2/help/options.html">options</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  library(ARTP2)
  ## Create data frame containing paths of build-in PLINK files that are going to used as reference
  ## As an example, use chromosomes 20-22
  chr  &lt;- 20:22
  nchr &lt;- length(chr)

  # Define reference data
  fam &lt;- vector("character", nchr)
  bim &lt;- vector("character", nchr)
  bed &lt;- vector("character", nchr)
  for(i in 1:nchr){
    fam[i] &lt;- system.file("extdata", package = "ARTP2", paste("chr", chr[i], ".fam", sep = ""))
    bim[i] &lt;- system.file("extdata", package = "ARTP2", paste("chr", chr[i], ".bim", sep = ""))
    bed[i] &lt;- system.file("extdata", package = "ARTP2", paste("chr", chr[i], ".bed", sep = ""))
  }
  reference &lt;- data.frame(fam, bim, bed, stringsAsFactors = FALSE)

  # Pathways do not need to be the same for each population. 
  # For this example, the pathway will be the same
  pathway &lt;- system.file("extdata", package = "ARTP2", "pathway.txt.gz")

  # logistic regression is used in base model, thus ncases and ncontrols should be specified. 
  family &lt;- 'binomial'

  ## Objects for population 1
  study1     &lt;- system.file("extdata", package = "ARTP2", "study1.txt.gz")
  ncases1    &lt;- list(c(9580, 2591))
  ncontrols1 &lt;- list(c(53810, 3052))
  lambda1    &lt;- 1.10

  # Set the options for population 1. 
  #############################################
  # !!! delete.gene.files must be set to FALSE
  #############################################
  options1 &lt;- list(inspect.snp.n = 2, nperm = 1e3, nthread=1,
                maf = .01, HWE.p = 1e-6,  gene.R2 = .9, 
                id.str = "Pop1", out.dir=getwd(), 
                save.setup = FALSE, 
                delete.gene.files=FALSE, seed=123)

  #ret1 &lt;- sARTP(summary.files = study1, pathway, family, reference, lambda1, 
  #            ncases1, ncontrols1, options = options1)
 
  # For population 2
  study2     &lt;- system.file("extdata", package = "ARTP2", "study2.txt.gz")
  ncases2    &lt;- list(7638)
  ncontrols2 &lt;- list(54319)
  lambda2    &lt;- 1.08

  # Set the options for population 2. 
  #############################################
  # !!! delete.gene.files must be set to FALSE
  # A different seed is used
  #############################################
  options2 &lt;- list(inspect.snp.n = 2, nperm = 1e3, nthread=1,
                maf = .01, HWE.p = 1e-6,  gene.R2 = .9, 
                id.str = "Pop2", out.dir=getwd(), 
                save.setup = FALSE, 
                delete.gene.files=FALSE, seed=456)

  #ret2 &lt;- sARTP(summary.files = study2, pathway, family, reference, lambda2, 
  #             ncases2, ncontrols2, options = options2)

  # Run ARTP2 by merging population results
  #mrg &lt;- warm.start.multiPop.gene(list(ret1, ret2))
  #mrg$pathway.pvalue
  #mrg$gene.pvalue

</code></pre>

<hr /><div style="text-align: center;">[Package <em>ARTP2</em> version 1.2.32 <a href="00Index.html">Index</a>]</div>
</div>
</body></html>
